### PR TITLE
DOC: Examples of monotonic/non-monotonic label-based indexing

### PR DESCRIPTION
**Pull request by nileracecrew**
Just ran into this gotcha and almost filed it as a bug report. Added a few examples to fill in the section header.

### CODE REVIEW

File: doc/source/gotchas.rst [hunk 1]
```diff
@@ -242,6 +242,57 @@ Label-based slicing conventions
 Non-monotonic indexes require exact matches
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+If the index of a ``DataFrame`` is montonically increasing or decreasing, then the bounds
```
Comment (jorisvandenbossche) [inline doc/source/gotchas.rst]: typo in 'montonically'

File: doc/source/gotchas.rst [hunk 2]
```diff
@@ -242,6 +242,57 @@ Label-based slicing conventions
...
+    2  NaN  NaN
+    3  NaN  NaN
+    4  NaN  NaN                
+
+    # OK
+    >>> df_nonmonotonic_index.loc[2:4, :]
+         a    b
+    2  NaN  NaN
+    3  NaN  NaN
+    4  NaN  NaN
```
Comment (jorisvandenbossche) [inline doc/source/gotchas.rst]: This is included twice

File: doc/source/gotchas.rst [hunk 3]
```diff
@@ -242,6 +242,57 @@ Label-based slicing conventions
...
+    Empty DataFrame
+    Columns: [a, b]
+    Index: []
+
+On the other hand, if the index is not monotonic, then both slice bounds must be
+*unique* members of the index. 
+
+.. code-block:: python
+
+    >>> df_nonmonotonic_index = pd.DataFrame(index=[2,3,4,5,3], columns=['a', 'b'])
```
Comment (jorisvandenbossche) [inline doc/source/gotchas.rst]: maybe make this `index=[3,2,4, ...]` then the `.loc`

File: doc/source/gotchas.rst [hunk 4]
```diff
@@ -242,6 +242,57 @@ Label-based slicing conventions
...
+    3  NaN  NaN
+    4  NaN  NaN    
+    
+    # 0 is not in the index
+    >>> df_nonmonotonic_index.loc[0:4, :]
+    Traceback (most recent call last):
+    ...
+    KeyError: 0
+    
+    >>> df_nonmonotonic_index.loc[2:3, :]
```
Comment (jorisvandenbossche) [inline doc/source/gotchas.rst]: maybe add here a comment like for the others "3 is not a unique label in the index"

**Comment by jorisvandenbossche** [comment]
@nileracecrew Thanks for this nice contribution to the docs! (that title was indeed missing an explanation ;-))

It is certainly good to keep it here, but we maybe should also consider to include a similar explanation in the indexing docs itself.

**Comment by nileracecrew** [comment]
You're welcoem @jorisvandenbossche. Seems like this behavior is most likely to show up when you have other problems (in my case corrupted data, as well as an in another issue I found), but it is surprising. 

I made the suggested edits and re-committed.

**Comment by jreback** [comment]
instead a empty frame use a range for number (range(6))

**Comment by nileracecrew** [comment]
Sorry, I don't quite understand your meaning. Do you mean fill the example DataFrame with numbers, or use range(6) as an index instead of the empty DataFrame example, or something else?

**Comment by jreback** [comment]
fill with numbers

**Comment by gfyoung** [comment]
@nileracecrew : If I'm not mistaken (unless the tests do check these files), since your fixes are purely documentation related, you should put <b>[ci skip]</b> in your commit message because you aren't changing any code.  Every time you make a change to your PR, you are adding another build to Travis (you currently have five).  Unless these `.rst` files are checked, I would suggest if someone could cancel all of your builds and you could put that "ci skip" tag in your commit.

File: doc/source/gotchas.rst [hunk 5]
```diff
@@ -242,6 +242,54 @@ Label-based slicing conventions
 Non-monotonic indexes require exact matches
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+If the index of a ``DataFrame`` is monotonically increasing or decreasing, then the bounds
```
Comment (jreback) [inline doc/source/gotchas.rst]: this should be in ``indexing.rst``

File: doc/source/gotchas.rst [hunk 6]
```diff
@@ -242,6 +242,54 @@ Label-based slicing conventions
...
+of a label-based slice can be outside the range of the index, much like slice indexing a 
+normal Python ``list``.
+
```
Comment (jreback) [inline doc/source/gotchas.rst]: hmm actually what I mean is we need to more references to this section as this is certainly easy to miss. maybe can do that in antother PR though.

File: doc/source/gotchas.rst [hunk 7]
```diff
@@ -242,6 +242,54 @@ Label-based slicing conventions
...
+.. code-block:: python
+
+   >>> df_monotonic_index = pd.DataFrame(index=[2,3,3,4,5], columns=['data'], data=range(5))
+    
+   # no rows 0 or 1, but still returns rows 2, 3 (both of them), and 4:
+   >>> df_monotonic_index.loc[0:4, :]
```
Comment (jreback) [inline doc/source/gotchas.rst]: actually make these all ipython blocks rather than code-blocks.

File: doc/source/gotchas.rst (hunk 5 above)
Comment (jreback) [inline doc/source/gotchas.rst]: Series or DataFrame

File: doc/source/gotchas.rst [hunk 8]
```diff
@@ -242,6 +242,54 @@ Label-based slicing conventions
 Non-monotonic indexes require exact matches
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+If the index of a ``DataFrame`` is monotonically increasing or decreasing, then the bounds
+of a label-based slice can be outside the range of the index, much like slice indexing a
```
Comment (jreback) [inline doc/source/gotchas.rst]: show the use of ``df_monotonic_index.index.is_monotonic``

**Comment by jreback** [comment]
pls squash when you update

**Comment by nileracecrew** [comment]
Is there a quick way to render a single .rst file to see the output? GitHub doesn't render the ipython blocks.

**Comment by jreback** [comment]
http://pandas-docs.github.io/pandas-docs-travis/contributing.html#how-to-build-the-pandas-documentation

File: doc/source/gotchas.rst (hunk 7 above)
Comment (jorisvandenbossche) [inline doc/source/gotchas.rst]: In this case it makes sense to use code blocks I think, as below there are some error messages shown (and they don't need to be shown fully). 
 But, maybe use IPython prompts for uniformity with the rest of the document?

**Comment by nileracecrew** [comment]
sorry, didn't mean to push that here

**Comment by jreback** [comment]
thanks!

**Pull request closed without merging**